src/cli/self_update/windows.rs

```diff
@@ -41,10 +48,102 @@ pub(crate) fn do_msvc_check(opts: &InstallOpts<'_>) -> bool {
+        let message = match self.0 {
+            740 => "elevation required",
+            1001 => "Visual Studio installer process is running",
+            1003 => "Visual Studio is in use",
+            1602 => "operation was canceled",
+            1618 => "another installation running",
+            1641 => "operation completed successfully, and reboot was initiated",
+            3010 => "operation completed successfully, but install requires reboot before it can be used",
+            5003 => "bootstrapper failed to download installer",
+            5004 => "operation was canceled",
+            5005 => "bootstrapper command-line parse error",
+            5007 => "operation was blocked - the computer does not meet the requirements",
+            8001 => "arm machine check failure",
+            8002 => "background download precheck failure",
+            8003 => "out of support selectable failure",
+            8004 => "target directory failure",
+            8005 => "verifying source payloads failure",
+            8006 => "Visual Studio processes running",
+            -1073720687 => "connectivity failure",
+            -1073741510 => "Microsoft Visual Studio Installer was terminated",
+            _ => "error installing Visual Studio"
+        };
+        write!(f, "{} (exit code {})", message, self.0)
     }
+}
 
-    true
+pub(crate) fn try_install_msvc() -> Result<()> {
+    // download the installer
+    let visual_studio_url = utils::parse_url("https://aka.ms/vs/17/release/vs_community.exe")?;
+
+    let tempdir = tempfile::Builder::new()
+        .prefix("rustup-visualstudio")
+        .tempdir()
+        .context("error creating temp directory")?;
+
+    let visual_studio = tempdir.path().join("vs_setup.exe");
+    utils::download_file(&visual_studio_url, &visual_studio, None, &|_| ())?;
+
+    // Run the installer. Arguments are documented at:
+    // https://docs.microsoft.com/en-us/visualstudio/install/use-command-line-parameters-to-install-visual-studio
+    let mut cmd = Command::new(visual_studio);
+    cmd.arg("--wait")
+        // Display an interactive GUI focused on installing just the selected components.
+        .arg("--focusedUi")
+        // Add the linker and C runtime libraries.
+        .args(["--add", "Microsoft.VisualStudio.Component.VC.Tools.x86.x64"]);
+
+    // It's possible an earlier or later version of the Windows SDK has been
+    // installed separately from Visual Studio so installing it can be skipped.
+    let mut has_libs = false;
+    if let Some(paths) = process().var_os("lib") {
+        for mut path in split_paths(&paths) {
+            path.push("kernel32.lib");
+            if path.exists() {
+                has_libs = true;
+            }
+        }
+    };
+    if !has_libs {
+        cmd.args([
+            "--add",
+            "Microsoft.VisualStudio.Component.Windows11SDK.22000",
```

Comment:
Is this the lowest common denominator?  I don't have Windows 11, would this cause an SDK which would build stuff which wouldn't run on my Windows PC?

Reply:
The latest SDK can be used to build for older versions. From [their website](https://developer.microsoft.com/en-us/windows/downloads/windows-sdk/) (my bold): 
> The Windows SDK (10.0.22000) for Windows 11 provides the latest headers, libraries, metadata, and tools for building Windows applications. Use this SDK to build Universal Windows Platform (UWP) and Win32 applications for Windows 11 and **previous Windows releases**.

Reply:
OK, thanks for the clarification.